forge-test/pegout/LpRefund.t.sol

```diff
@@ -0,0 +1,617 @@
+            quoteHash,
+            btcTx,
+            BLOCK_HEADER_HASH,
+            PARTIAL_MERKLE_TREE,
+            merkleHashes
+        );
+    }
+
+    function test_RefundPegOut_RevertsIfQuoteWasNotPaid() public {
+        Quotes.PegOutQuote memory quote = createTestPegOutQuote(
+            1 ether,
+            pegOutLp
+        );
+        bytes32 quoteHash = pegOutContract.hashPegOutQuote(quote);
+
+        // Don't deposit - try to refund directly
+        bytes memory btcTx = generateBtcTx(quote, quoteHash);
+
+        vm.prank(pegOutLp);
+        vm.expectRevert(
+            abi.encodeWithSelector(Flyover.QuoteNotFound.selector, quoteHash)
+        );
+        pegOutContract.refundPegOut(
+            quoteHash,
+            btcTx,
+            BLOCK_HEADER_HASH,
+            PARTIAL_MERKLE_TREE,
+            merkleHashes
+        );
+    }
+
+    function test_RefundPegOut_RevertsIfNotCalledByLP() public {
+        Quotes.PegOutQuote memory quote = createAndDepositQuote();
+        bytes32 quoteHash = pegOutContract.hashPegOutQuote(quote);
+
+        bytes memory btcTx = generateBtcTx(quote, quoteHash);
+
+        // fullLp tries to refund pegOutLp's quote
+        vm.prank(fullLp);
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                Flyover.InvalidSender.selector,
+                pegOutLp,
+                fullLp
+            )
+        );
+        pegOutContract.refundPegOut(
+            quoteHash,
+            btcTx,
+            BLOCK_HEADER_HASH,
+            PARTIAL_MERKLE_TREE,
+            merkleHashes
+        );
+    }
+
+    function test_RefundPegOut_RevertsIfBtcTxNotRelatedToQuote() public {
+        Quotes.PegOutQuote memory quote = createAndDepositQuote();
+        bytes32 quoteHash = pegOutContract.hashPegOutQuote(quote);
+
+        // Create a different quote and generate tx for it (different hash in OP_RETURN)
+        Quotes.PegOutQuote memory otherQuote = createTestPegOutQuote(
+            0.5 ether,
+            pegOutLp
+        );
+        bytes32 otherQuoteHash = pegOutContract.hashPegOutQuote(otherQuote);
+
+        // Generate BTC tx with the OTHER quote's hash
+        bytes memory btcTx = generateBtcTx(quote, otherQuoteHash); // Wrong hash!
+
+        vm.prank(pegOutLp);
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                IPegOut.InvalidQuoteHash.selector,
+                quoteHash,
+                otherQuoteHash
+            )
+        );
+        pegOutContract.refundPegOut(
+            quoteHash,
+            btcTx,
+            BLOCK_HEADER_HASH,
+            PARTIAL_MERKLE_TREE,
+            merkleHashes
+        );
+    }
+
+    function test_RefundPegOut_RevertsIfNullDataMalformed() public {
+        Quotes.PegOutQuote memory quote = createAndDepositQuote();
+        bytes32 quoteHash = pegOutContract.hashPegOutQuote(quote);
+
+        // Test that a malformed Bitcoin transaction (too short) reverts
```

Comment:
In this case I would rename the test, this particular name referes a cases where the content of the null script is not the one we expect